File: backend/src/main/java/com/edu/management/entity/Student.java
package com.edu.management.entity;

import com.edu.management.enums.StudentStatus;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "students")
public class Student {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 50)
    private String name;
    
    @Column(length = 50)
    private String parentName;
    
    @Column(nullable = false, length = 20)
    private String parentPhone;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StudentStatus status = StudentStatus.ACTIVE;
    
    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: mobile-app/src/api/student.ts
import { get } from '../utils/request'

export interface Student {
  id: number
  name: string
  parentPhone?: string
  gradeLevel: string
  status: 'ACTIVE' | 'INACTIVE'
}

// 获取所有学生
export const getAllStudents = () => {
  return get<Student[]>('/students')
}

// 获取班级学生
export const getStudentsByClass = (classId: number) => {
  return get<Student[]>(`/students/class/${classId}`)
}
